Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs

```diff
@@ -108,42 +108,46 @@ IEnumerable<PropertyDeclarationSyntax> PerMethod(MethodDetails method)
                 var cSync = XmlDoc.C($"{_svc.ClientAbstractTyp.Name}.{method.SyncMethodName}");
                 var cAsync = XmlDoc.C($"{_svc.ClientAbstractTyp.Name}.{method.AsyncMethodName}");
                 // Add the general per-method settings property.
-                var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true);
+                // We always specify an expiration, and the retry is optional.
+                var expiration = method.Expiration ?? Expiration.None;
+                var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true)
+                    .WithInitializer(_ctx.Type<CallSettings>().Call(nameof(CallSettings.FromExpiration))(
```

Comment:
This initializer looks like it's overridden if the method has retry.
Would it be better to only apply this initializer in the `else` clause a bit further down?

Reply:
It's not so much overridden as built on. But I can probably do something cleaner.